src/webviews/main.ts

```diff
@@ -78,63 +312,243 @@ export async function createVueWebview<TRequest, TResponse>(params: WebviewParam
+        webviewJs: params.webviewJs,
+        cspSource: updateCspSource(webview.cspSource),
+    })
+
+    return webview
+}
+
+/**
+ * Resolves the webview HTML based off whether we're running from a development server or bundled extension.
+ */
+function resolveWebviewHtml(params: {
+    scripts: string
+    stylesheets: string
+    cspSource: string
+    webviewJs: string
+    main: vscode.Uri
+}): string {
+    const resolvedParams = { ...params, connectSource: 'none' }
+    const LOCAL_SERVER = process.env.WEBPACK_DEVELOPER_SERVER
+
+    if (LOCAL_SERVER) {
+        const local = vscode.Uri.parse(LOCAL_SERVER)
+        resolvedParams.cspSource = `${params.cspSource} ${local.toString()}`
+        resolvedParams.main = local.with({ path: `/${params.webviewJs}` })
+        resolvedParams.connectSource = `'self' ws:`
+    }
 
-    view.title = params.name
-    view.webview.html = `<html>
+    return `<html>
     <head>
         <meta charset="UTF-8">
         <meta name="viewport" content="width=device-width, initial-scale=1.0">
-        
+
         <meta
             http-equiv="Content-Security-Policy"
             content=
                 "default-src 'none';
-                img-src ${view.webview.cspSource} https:;
-                script-src ${view.webview.cspSource};
-                style-src ${view.webview.cspSource} 'unsafe-inline';
+                connect-src ${resolvedParams.connectSource};
+                img-src ${resolvedParams.cspSource} https:;
+                script-src ${resolvedParams.cspSource};
+                style-src ${resolvedParams.cspSource} 'unsafe-inline';
                 font-src 'self' data:;"
         >
     </head>
     <body>
         <div id="vue-app"></div>
         <!-- Dependencies -->
-        ${scripts}
-        ${stylesheets}
+        ${resolvedParams.scripts}
+        ${resolvedParams.stylesheets}
         <!-- Main -->
-        <script src="${mainScript}"></script>
+        <script src="${resolvedParams.main}"></script>
     </body>
 </html>`
+}
 
-    if (params.initialCalls) {
-        view.webview.onDidReceiveMessage((message: any) => {
-            if (message.command === 'initialized') {
-                for (const call of params.initialCalls!) view.webview.postMessage(call)
-            }
-        })
+/**
+ * Updates the CSP source for webviews with an allowed source for AWS endpoints when running in
+ * Cloud9 environments. Possible this can be further scoped to specific C9 CDNs or removed entirely
+ * if C9 injects this.
+ */
+export function updateCspSource(baseSource: string) {
+    return isCloud9() ? `https://*.amazonaws.com ${baseSource}` : baseSource
+}
+
+/**
+ * To preserve compatability with our min-version types we are declaring only portions of the latest API.
+ *
+ * Introduced in VS Code 1.50.0. Types sourced from:
+ * https://github.com/DefinitelyTyped/DefinitelyTyped/blob/ccaaa6ab6a4168c8644fd3c391c8bff9e485734b/types/vscode/index.d.ts
+ */
+declare module 'vscode' {
```

Comment:
Types for `WebviewView` are added via declaration merging. This is perfectly safe since it's just a TS construct. The actual logic checks the version and does not attempt to use the API if the version is too low, logging a warning message.